lib/OpenQA/Command.pm

```diff
@@ -81,14 +83,19 @@ sub parse_headers {
 }
 
 sub parse_params {
-    my ($self, @args) = @_;
+    my ($self, $args, $param_file) = @_;
 
     my %params;
-    for my $arg (@args) {
-        next unless $arg =~ /^([[:alnum:]_\[\]\.]+)=(.*)$/s;
+    for my $arg (@{$args}) {
+        next unless $arg =~ $PARAM_RE;
         push @{$params{$1}}, $2;
     }
 
+    for my $arg (@{$param_file}) {
+        next unless $arg =~ $PARAM_RE;
+        push @{$params{$1}}, path($2)->slurp;
```

Comment:
Actually, this would probably more efficient as `push @{$params{$1}}, {file => $2};`, since it allows `Mojo::UserAgent` to stream the file in chunks, instead of loading everything into memory.

Reply:
That doesn't seem to work as expected:
```
# openqa-cli api --host http://mango.qa.suse.de --apikey **** --apisecret **** -X POST \
    job_templates_scheduling/2 schema=JobTemplates-01.yaml preview=1 \
    --param-file template=sle15_sp3_ha.yml
400 Bad Request
{"error":["Job template name 'create_hdd_ha_textmode' with sle-15-SP3-Online-aarch64 and aarch64 is already used in job group 'HA'"],"error_status":400,"id":2,"job_group_id":2}
# vi ./lib/OpenQA/Command.pm
# openqa-cli api --host http://mango.qa.suse.de --apikey **** --apisecret **** -X POST \
    job_templates_scheduling/2 schema=JobTemplates-01.yaml preview=1 --param-file template=sle15_sp3_ha.yml
400 Bad Request
{"error":[{"message":"Expected object - got string.","path":"\/"}],"error_status":400}
```
Perhaps I am mistaken, but don't we require to load the whole file into memory before submitting the request to the server?

Reply:
By the way, this is actually still a valid point for improvement. Not sure how to make it work exactly within Mojolicious but generally it would be preferable to avoid reading the whole file into memory.

Reply:
I think i see now why it didn't work. The current implementation of `{file => '...'}` forces it to be a file upload with filename. Which doesn't work with the server side validation we have that expects a simple form value.

Reply:
Although the memory usage of the YAML parser on the server-side might just be the next thing to explode then :-)

Reply:
I've made an upstream change in Mojolicious to allow this in the future. https://github.com/mojolicious/mojo/compare/529af01cdb4d...884d8f1ee196